search-service/src/main/java/com/movie/recommender/search/service/SearchService.java

```diff
@@ -0,0 +1,140 @@
+            query.addCriteria(Criteria.where("rating")
+                    .gte(filter.getRatingStart())
+                    .lte(filter.getRatingEnd()));
+            log.info("Filtering by rating range: {} - {}", filter.getRatingStart(), filter.getRatingEnd());
+        }
+
+        if (filter.getIsAdult() != null) {
+            query.addCriteria(Criteria.where("adult").is(filter.getIsAdult()));
+            log.info("Filtering by adult content: {}", filter.getIsAdult());
+        }
+
+        if (filter.getCastNames() != null && !filter.getCastNames().isEmpty()) {
+            query.addCriteria(Criteria.where("cast.name").all(filter.getCastNames()));
+            log.info("Filtering by cast names: {}", filter.getCastNames());
+        }
+
+        if (filter.getCrewNames() != null && !filter.getCrewNames().isEmpty()) {
+            query.addCriteria(Criteria.where("crew.name").all(filter.getCrewNames()));
+            log.info("Filtering by crew names: {}", filter.getCrewNames());
+        }
+
+        if (filter.getKeywords() != null && !filter.getKeywords().isEmpty()) {
+            query.addCriteria(Criteria.where("keywords.name").all(filter.getKeywords()));
+            log.info("Filtering by keywords: {}", filter.getKeywords());
+        }
+
+        List<MovieDetails> result = mongoTemplate.find(query, MovieDetails.class);
+        log.info("Query executed, found {} movies", result.size());
+        return result;
+    }
+
+    private void validateRating(Double ratingStart, Double ratingEnd) {
+        if (ratingStart > ratingEnd) {
+            throw new IllegalArgumentException("ratingStart must be lower than ratingEnd");
+        }
+        if (ratingStart < MINIMAL_RATING) {
+            throw new IllegalArgumentException("ratingStart must be greater than " + MINIMAL_RATING);
+        }
+        if (ratingEnd > MAXIMAL_RATING) {
+            throw new IllegalArgumentException("ratingEnd must be lower than " + MAXIMAL_RATING);
+        }
+    }
+
+    private void validateDate(String startDate, String endDate) {
+        if (startDate.trim().isEmpty() || endDate.trim().isEmpty()) {
+            throw new IllegalArgumentException("Dates cannot be empty");
+        }
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        try {
+            LocalDate dateStart = LocalDate.parse(startDate, formatter);
+            LocalDate dateEnd = LocalDate.parse(endDate, formatter);
+
+            if (dateStart.isAfter(dateEnd)) {
+                throw new IllegalArgumentException("startDate must be before endDate");
+            }
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format " + startDate + " and " + endDate);
+        }
+    }
+
+    private void validateGenres(List<String> genres) {
+        for (String genre : genres) {
+            if (!DEFAULT_GENRES.contains(genre)) {
```

Comment:
make this comparison case insensitive so that `Drama` and `drama` are treated as the same and pass the validation
additionally, it would be better to store the default genres in a properties file rather than hardcoding them in the code